addons/mail/static/src/utils/test_utils.js

```diff
@@ -169,47 +170,99 @@ function nextAnimationFrame() {
+    if (response.status !== 200) {
+        throw new Error('Error while fetching required models');
+    }
+    modelDefinitions = new Map(Object.entries(await response.json()));
+
+    for (const [modelName, fields] of modelDefinitions) {
+         // insert fields present in the fieldsToInsert registry : if the field
+         // exist, update it's default value according to the one in the
+         // registry; If it does not exist, add it to the model definition.
+        const fieldNamesToFieldToInsert = fieldsToInsertRegistry.category(modelName).getEntries();
+        for (const [fname, fieldToInsert] of fieldNamesToFieldToInsert) {
+            if (fname in fields) {
+                fields[fname].default = fieldToInsert.default;
+            } else {
+                fields[fname] = fieldToInsert;
+            }
+        }
+        // apply default values for date like fields if none was passed
+        for(const fname in fields) {
+            const field = fields[fname];
+            if (['date', 'datetime'].includes(field.type) && !field.default) {
+                const defaultFieldValue =
+                    field.type === 'date'
+                    ? () => moment().format('YYYY-MM-DD')
+                    : () => moment.utc().format("YYYY-MM-DD HH:mm:ss")
+                field.default = defaultFieldValue;
+            }
+        }
+    }
+}
 
-    data.publicPartnerId = 4;
-    data['res.partner'].records.push({
-        active: false,
-        display_name: "Public user",
-        id: data.publicPartnerId,
-    });
-    data.publicUserId = 3;
-    data['res.users'].records.push({
-        active: false,
-        display_name: "Public user",
-        id: data.publicUserId,
-        name: "Public user",
-        partner_id: data.publicPartnerId,
-    });
+//------------------------------------------------------------------------------
+// Public: test lifecycle
+//------------------------------------------------------------------------------
+
+async function beforeEach(self) {
+    if (!modelDefinitions) {
+        await setupModelDefinitions();
+    }
+    const data = { ...TEST_USER_IDS };
+    const recordsToInsertRegistry = registry.category('mail.model.definitions').category('recordsToInsert');
+    for (const [modelName, fields] of modelDefinitions) {
+        let records = [];
+        if (recordsToInsertRegistry.contains(modelName)) {
+            records = JSON.parse(JSON.stringify(recordsToInsertRegistry.get(modelName)));
+        }
+        data[modelName] = { fields, records };
+    }
+    // some tests rely on this fake model, all fake models will be removed in another PR
+    data['res.fake'] = {
+        fields: {
+            activity_ids: { string: "Activities", type: 'one2many', relation: 'mail.activity' },
+            email_cc: { type: 'char' },
+            partner_ids: { relation: 'res.partner', string: "Related partners", type: 'many2one', },
+        },
+        records: [],
+    };
```

Comment:
`beforeEach` is kinda deprecated, possible to do it in start instead?

Reply:
It seems difficult. Indeed, tests use before each which puts `data` into `this` then use those model definitions before calling `start`... The scope of this PR is to replace fake model definitions by the one on the server. The other PR (the one with `pyEnv`) will allow us to remove `beforeEach` calls so I think it's better to wait for this other PR.

Reply:
ok for later